app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioOption.kt

```diff
@@ -0,0 +1,454 @@
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            modifier = Modifier.weight(1f),
+            color = optionTextColor,
+          )
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(state)
+        }
+        is Label -> {
+          Column(Modifier.weight(1f)) {
+            HedvigText(
+              optionText,
+              style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+              color = optionTextColor,
+            )
+            HedvigText(
+              radioOptionStyle.labelText,
+              style = radioOptionSize.size(radioOptionStyle).labelTextStyle,
+              color = labelTextColor,
+            )
+          }
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(state)
+        }
+        LeftAligned -> {
+          SelectIndicationCircle(state)
+          Spacer(Modifier.width(8.dp))
+          HedvigText(
+            optionText,
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            color = optionTextColor,
+            modifier = Modifier.weight(1f),
+          )
+        }
+      }
+    }
+  }
+}
+
+sealed interface IconResource {
+  data class Vector(val imageVector: ImageVector) : IconResource
+
+  data class Painter(val painterResId: Int) : IconResource
+}
+
+@Composable
+fun SelectIndicationCircle(state: RadioOptionState, modifier: Modifier = Modifier) {
+  CheckItemAnimation(state) { currentState: RadioOptionState ->
+    Box(
+      contentAlignment = Alignment.Center,
+    ) {
+      Spacer(
+        modifier
+          .size(24.dp)
+          .clip(CircleShape)
+          .then(
+            when (currentState) {
+              Chosen -> Modifier.border(8.dp, radioOptionColors.chosenIndicatorColor, CircleShape)
+              NotChosen -> Modifier.border(2.dp, radioOptionColors.notChosenIndicatorColor, CircleShape)
+              ChosenLocked -> Modifier.border(8.dp, radioOptionColors.disabledIndicatorColor, CircleShape)
+            },
+          ),
+      )
+    }
+  }
+}
+
+@Composable
+private fun CheckItemAnimation(state: RadioOptionState, content: @Composable (state: RadioOptionState) -> Unit) {
+  val selectedTransition = updateTransition(state)
+  selectedTransition.AnimatedContent(
+    transitionSpec = {
+      fadeIn(tween(durationMillis = 200, easing = EaseInOut)) togetherWith fadeOut(tween(200, easing = EaseInOut))
+    },
+    contentAlignment = Alignment.Center,
+  ) { currentState ->
+    content(currentState)
+  }
+}
+
+object RadioOptionDefaults {
+  internal val radioOptionStyle: RadioOptionStyle = Default
+  internal val radioOptionSize: RadioOptionSize = RadioOptionSize.Large
+
+  sealed interface RadioOptionStyle {
+    data object Default : RadioOptionStyle
+
+    data class Label(val labelText: String) : RadioOptionStyle
+
+    data class Icon(val iconResource: IconResource) : RadioOptionStyle // we need another solution here. Content?
```

Comment:
Another solution for what, if there is a scenario where we need both an icon and a label?

I think there's two approaches to consider here.
If the component really does not mix and match those this sealed interface should be a simple API which lets us easily just do `RadioOptionStyle.` and get autocompletion for all the relevant styles.

If we figure out that we want both, we could have the icon not be part of RadioOptionStyle, but be a normal parameter in the public `RadioOption` function. Like `iconResource: IconResource? = null`.
And then on the caller, if you pass nothing you get nothing, if you pass non-null the icon should show and the UI should adjust accordingly.

It's a question of how it's gonna be used really to know what the best option is. We can also redesign that later if we find that we need that shape instead.

Reply:
Oh I do see now that the left-aligned one would simply not work with an Icon really. Maybe we should not allow the icon in all scenarios then.

Reply:
I say let's keep it like this here, and if the need for this combination arises, later add 
`data class LabelAndIcon(val labelText: String, val iconResource: IconResource) : RadioOptionStyle`